### DISCUSSION TITLE
Deleting an M2A record should remove it from the model using it

### DISCUSSION THREAD

**Discussion by notflip** [directus/directus#21485] [Feature Requests]
### Summary

Let's say we have a **Page** model with a builder element called blocks.

Page

- blocks (M2A)

Page_blocks

- pages_id
- item
- collection

block_images

- id
- image

---

If I have a page, which has the block_images block attached, and I delete the block_images from Directus, that page will still have the block_images record in it's blocks.

Would it be possible to have the possibility to delete any rows using a certain (M2A) block, when that block is deleted?

### Motivation

If a block is deleted now, we end up with empty implementations of that block in all pages, which cause issues with missing data in the frontend.

### Detailed Design

 No comment

### Must Have:
- The possibility to CASCADE implementations of a certain block M2A

### Drawbacks

No comment

### Alternatives

No comment

### Adoption Strategy

No comment

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
